Keyword `minItems` (#126)
Intelligence-AI/learnjsonschema.com: content/2020-12/validation/minItems.markdown

  - vocabulary: applicator
    keyword: contains
---

The `minItems` keyword specifies the minimum number of items that must be present in an array. It can be used to define constraints on the size of an array, ensuring that it contains at least a certain number of elements.
* Applies to arrays only.
* Value must be a non-negative integer.
* An array is valid if it has at least the specified number of elements.
* Omitting `minItems` keyword has the same behavior as a value of 0.

{{<alert>}}

> _**Note:**  `items`, `prefixItems`, and `contains` do not affect `minItems`. The instance is first evaluated against `minItems` (if present), and if it validates successfully, it is further evaluated against `items`, `prefixItems`, and `contains`._

Reviewer: ```suggestion
_**Note:**  `items`, `prefixItems`, and `contains` do not affect `minItems`. The instance independently evaluated against `minItems` (if present)
```

JSON Schema doesn't introduce any actual ordering here, it just says that they are independent

{{</alert>}}

## Examples

{{<schema `Schema with 'minItems' keyword`>}}
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "type": "array",
  "minItems": 3
}
{{</schema>}}

{{<instance-pass `An array instance with 3 or more elements is valid`>}}
[ 1, true, "hello" ]
{{</instance-pass>}}

{{<instance-fail `An array instance with less than 3 elements is invalid`>}}
[ 1, "apple" ]
{{</instance-fail>}}

{{<schema `Schema with 'minItems' keyword`>}}
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "type": "array",

>   "minItems": -2

Reviewer: This schema is not valid. I don't think we are showing invalid schemas for other keywords, so maybe let's leave this out?
